src/test/ripple-libpp_demo.cpp

```diff
@@ -178,7 +171,189 @@ int main (int argc, char** argv)
+    }};
+
+    std::cout << "\nBefore signing: \n"
+        << noopTx.getJson(0, false).toStyledString()  << std::endl;
+
+    return noopTx;
+}
+
+// Apply one multi-signature to the supplied transaction.  The signer
+// provides their AccountID, PublicKey, and SecretKey.
+bool multisign (ripple::STTx& tx, Credentials const& signer)
+{
+    using namespace ripple;
+
+    // Get the TxnSignature.
+    Serializer s = buildMultiSigningData (tx, signer.id());
+
+    auto const multisig = ripple::sign (
+        signer.publicKey(), signer.secretKey(), s.slice());
+
+    // Make the signer object that we'll inject into the array.
+    STObject element (sfSigner);
+    element[sfAccount] = signer.id();
+    element[sfSigningPubKey] = signer.publicKey();
+    element[sfTxnSignature] = multisig;
+
+    // If a Signers array does not yet exist make one.
+    if (! tx.isFieldPresent (sfSigners))
+        tx.setFieldArray (sfSigners, {});
+
+    // Insert the signer into the array.
+    STArray& signers {tx.peekFieldArray (sfSigners)};
+    signers.emplace_back (std::move (element));
+
+    // Sort the Signers array by Account.  If it is not sorted when submitted
+    // to the network then it will be rejected.
+    std::sort (signers.begin(), signers.end(),
+        [](STObject const& a, STObject const& b)
+    {
+        return (a[sfAccount] < b[sfAccount]);
+    });
+
+    // Verify that the signature is valid.
+    bool const pass = tx.checkSign(true).first;
+    assert (pass);
+
+    // To submit multisigned JSON to the network use this RPC command:
+    // $ rippled submit_multisigned '<all JSON>'
+    std::cout << "\nMultisigned JSON: \n"
+        << tx.getJson(0, false).toStyledString()  << std::endl;
+
+    // Alternatively, to submit the multisigned blob to the network:
+    //  1. Extract the hex string (including the quotes) following "tx"
+    //  2. Then use this RPC command:
+    //     $ rippled submit <quoted hex string>
+    std::cout << "Multisigned blob:"
+        << tx.getJson(0, true) << std::endl;
+
+    return pass;
+}
+
+// Outlines a multisign process where:
+//  1. A multi-signable transaction is built.
+//  2. That transaction is signed by one signer.
+//  3. The transaction is signed by a different signer.
+bool exerciseMultisign()
+{
+    using namespace ripple;
+
+    // Create credentials for the folks involved in the transaction.
+    Credentials const alice {"alice"};
+    Credentials const billy {"billy"};
+    Credentials const carol {"carol"};
+
+    // Create a transaction on alice's account.  alice doesn't sign it.
+    STTx tx {buildMultisignTx (alice.id(), 2, 100)};
+
+    // billy and carol sign alice's transaction for her.
+    bool allPass = multisign (tx, billy);
+    allPass &= multisign (tx, carol);
+
+    return allPass;
+}
+
+//------------------------------------------------------------------------------
+
+int main (int argc, char** argv)
+{
+#if defined(__GNUC__) && !defined(__clang__)
```

Comment:
I'm just as happy to leave the version checks here where I can understand them.  But if @ximinez also thinks they should move, I can move 'em.  They are the same checks as were originally in the file.  They just look new because of where the added code went.